CorsixTH/Lua/entities/humanoid.lua

```diff
@@ -491,7 +491,8 @@ function Humanoid:startAction()
   -- Call the action start handler
   TheApp.humanoid_actions[action.name](action, self)
 
-  if action == self.action_queue[1] and action.todo_interrupt then
+  if action == self.action_queue[1] and action.todo_interrupt and not action.uninterruptible then
+    -- Interrupt an action because it has been marked as to be interrupted
```

Comment:
```suggestion
    -- Interrupt the current action.
```
The first action in an action queue is currently being done, so it's "the" action rather than "an (any/some) action".

Inside the `if` the decision is already made, so why it's here isn't relevant anymore. Just state what is going to happen now.

The reason for not stating why it's here: All code is changing all the time, so eventually the "why" explanation becomes incomplete or even incorrect.
In other words, avoid duplicating information.

Reply:
> The first action in an action queue is currently being done, so it's "the" action rather than "an (any/some) action".
> 
> Inside the `if` the decision is already made, so why it's here isn't relevant anymore. Just state what is going to happen now.
> 
> The reason for not stating why it's here: All code is changing all the time, so eventually the "why" explanation becomes incomplete or even incorrect. In other words, avoid duplicating information.

I didn't understand and didn't know that `startAction()` would delete actions out of the queue for some reason.
It's not obvious.
It was a bit of a surprise that `startAction()` does this.
This is exactly what I wanted to make more transparent in the code.

Perhaps I shouldn't have written this inside `if`, and it should be outside `if`.
And perhaps it should be written in the description of the function itself.

In general, the point is not only to explain what happens inside `if`.
But to explain why `startAction()` does this.

Perhaps this needs to be refactered. 
But it would be better to at least simply explain why this is so.

Reply:
Documenting is fine.

The point I am trying to make is that you your comments should always be about the code *below* it. If you want to say something about the code above, write a comment about that code at a line earlier in the file.

In that way, a reader doesn't have to look up and down the file, trying to find a comment that explains the line of code that gives trouble.

In your change, the "because it has been marked as to be interrupted" part is about the `if` statement above it. It's not about the code below (the code below just aborts blindly without caring why it must do that).  
It's fine that you want to state what the `if` is doing, but write that text above the `if` line.

Here, you could write above line 494 `-- Check whether the first action is interrupted.` (Instead if `first` you could also write `current`.)

Then at 495 you could write a short comment like `-- Abort the action.` or so.

Reply:
> Here, you could write above line 494 `-- Check whether the first action is interrupted.` (Instead if `first` you could also write `current`.)
> 
> Then at 495 you could write a short comment like `-- Abort the action.` or so.

OK, thank you.